src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -132,17 +75,81 @@ public void doPost(HttpServletRequest request, HttpServletResponse response) thr
       entity.setProperty("completion_date", StringUtil.escapeQuotesInParameter(completion_date));
       entity.setProperty("dependency", StringUtil.escapeQuotesInParameter(dependency));
       entity.setProperty("status", status);
+      entity.setProperty("scheduled_date", -1);
       datastore.put(entity);
-    
+      ListRecord newEntity = entityToLR(entity);
+      newEntity.createCalendarEvent();
     response.sendRedirect("/index.html");
   }
 
-  private String getParameter(HttpServletRequest request, String name, String defaultValue) 
+  private String getParameter(HttpServletRequest request, String name, String defaultValue)
   {
     String value = request.getParameter(name);
     if (value == null) {
       value = defaultValue;
     }
     return value;
   }
+
+  private ListRecord entityToLR(Entity entity) {
```

Comment:
This is a pretty lengthy function, if you could describe what's happening (getting a properties from the entity and assigning them to a variable, which is then used to create a ListRecord, that would be helpful!